tests/test_curve.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+from umbral.curve import Curve, SECP256R1, SECP256K1, SECP384R1
+
+
+def test_curve_whitelist():
+    # Test the AVAIL_CURVES dict to have only these three curves:
+    assert len(Curve._Curve__AVAIL_CURVES) == 3
+    assert Curve._Curve__AVAIL_CURVES['secp256r1'] == 415
+    assert Curve._Curve__AVAIL_CURVES['secp256k1'] == 714
+    assert Curve._Curve__AVAIL_CURVES['secp384r1'] == 715
+
+    # Test that we can't instantiate other curves:
+    with pytest.raises(ValueError):
+        Curve(711)
+
+    # Test the hardcoded curves are what they're supposed to be:
+    test_p256 = SECP256R1()
+    test_secp256k1 = SECP256K1()
+    test_p384 = SECP384R1()
+
+    # Test the supported curves property
```

Comment:
Eh?  What are we showing here?

Reply:
They're all caps because they act as constants. Nothing in the structure actually changes, there are no modifying methods.

If by importing the instances you mean creating a new instance with every import, I would disagree. This `Curve` object contains the `EC_GROUP` which is the primary cause of memory issues in pyUmbral. Instantiating it over and over again is the result of the memory leak just fixed.

Currently, we access the curve through the `Config` interface via the `default_curve` or `default_params` functions. This API still preserves this architecture, but I don't like it. I'm thinking that there should be some object that contains the params, the config, and the curve all in one.

Reply:
> They're all caps because they act as constants.

But... they don't.  The user can (intentionally or otherwise) init a bunch of them.  They're not constants - and why not?  Let's make them constants. 

> If by importing the instances you mean creating a new instance with every import,

Wait, what?  I didn't say anything of the sort.  I don't even know how you'd do that - with some importlib gymnastics or something?  Anyway, no I'm saying just make them instances instead of classes.